server/src/main/java/org/elasticsearch/common/logging/ESLogMessage.java

```diff
@@ -16,43 +16,82 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-
 package org.elasticsearch.common.logging;
 
 import com.fasterxml.jackson.core.io.JsonStringEncoder;
+import org.apache.logging.log4j.message.MapMessage;
 import org.apache.logging.log4j.message.ParameterizedMessage;
-import org.elasticsearch.common.SuppressLoggerChecks;
+import org.apache.logging.log4j.util.Chars;
+import org.apache.logging.log4j.util.StringBuilders;
 
 import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
 import java.util.Map;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 /**
  * A base class for custom log4j logger messages. Carries additional fields which will populate JSON fields in logs.
  */
-public abstract class ESLogMessage extends ParameterizedMessage {
+public class ESLogMessage extends MapMessage<ESLogMessage, Object> {
```

Comment:
`ESLoggerUsageTests` only tests for  `ParameterizedMessage` so we need to extend the testing.

Reply:
there is actually a lot of work needed to improve `ESLOgChecker` 
usages like `ESLogMessages.of("msg {}").field("arg","arg-value")` should be checked considered ok. but when arg value is not provided then we should raise exception.

Not entirely sure how to extend this yet